crates/sui/tests/reconfiguration_tests.rs

```diff
@@ -309,15 +313,191 @@ async fn test_validator_resign_effects() {
     assert_eq!(effects0.into_message(), effects1.into_message());
 }
 
+#[sim_test]
+async fn test_reconfig_with_committee_change_basic() {
+    // This test exercise the full flow of a validator joining the network, catch up and then leave.
+
+    let init_configs = test_and_configure_authority_configs(4);
```

Comment:
In order to better "test" this flow we probably want to set the validators to ignore all peer connections so that we can verify that new nodes joining can really "talk" with the other validators in the set....Although today this information isn't included on chain (their p2p network address) so that would need to be added later. 

I don't think we need to block this PR for the required changes but we should mark a TODO about it and ensure we fix it prior to mainnet.

Reply:
+1 on the TODO